editor/components/block-inspector/advanced-controls.js

```diff
@@ -30,25 +31,32 @@ class BlockInspectorAdvancedControls extends Component {
 	}
 
 	render() {
-		const { selectedBlock } = this.props;
-		const blockType = getBlockType( selectedBlock.name );
-		if ( false === blockType.className ) {
-			return null;
-		}
-
 		return (
-			<PanelBody
-				className="editor-advanced-controls"
-				initialOpen={ false }
-				title={ __( 'Advanced' ) }
-			>
-				{ false !== blockType.className &&
-					<InspectorControls.TextControl
-						label={ __( 'Additional CSS Class' ) }
-						value={ selectedBlock.attributes.className || '' }
-						onChange={ this.setClassName } />
-				}
-			</PanelBody>
+			<Slot
+				name="Inspector.AdvancedControls"
+				renderFills={ ( fills ) => {
```

Comment:
I think this is a valuable feature (to hook into rendering when we know that fills exist, or alternatively when no fills exist, see #1343).

Do you think it might be more simple to implement as a [Function-as-Child](https://medium.com/merrickchristensen/function-as-child-components-5f3920a9ace9) (a more common pattern)? 

```jsx
<Slot name="Inspector.AdvancedControls">
	{ ( fills ) => <div>{ fills }</div> }
</Slot>
```

Reply:
Yes, this is another way of doing it. Personally, I feel a bit awkward about treating `this.props.children` as a function. However, I'm fine with both patterns as long as they do their work :)